extensions/access_log_policy/plugin.cc

```diff
@@ -143,6 +147,26 @@ void PluginContext::onLog() {
   setFilterStateValue(false);
 }
 
+bool PluginContext::isRequestFailed() {
+  // Check if HTTP request is a failure.
+  int64_t http_response_code = 0;
+  getValue({kResponse, kCode}, &http_response_code);
+  // If request is a failure, log it.
+  if (http_response_code != 200) {
+    return true;
+  }
+
+  // Check if gRPC request is a failure.
+  int64_t grpc_response_code = ::Wasm::Common::kDefaultGrpcStatusCode;
+  getValue({kResponse, kGrpcStatus}, &grpc_response_code);
```

Comment:
Is the default value necessary? getValue should run non-ok if not found. Also the logic does not seem to be right. Shouldn't we check if response_code is not OK (0)?

Reply:
For gRPC , I am not sure why but OK is value 2...
and hence if getValue return non-ok and keep grpc_response_code as 0, we will think request is always a failure and even though it is not.

Reply:
@douglas-reid : I see you set  'grpc_status = 2', why is OK for gRPC value 2?

Reply:
I think Doug just want to make `unknown` the default, which is 2.

Reply:
yeah, i set it to `2` for `unknown`, but if you look at the code, if the protocol isn't `grpc` we leave it empty.

Reply:
Thanks.. changed logic to check for OK(0) and protocol check for gRPC too